bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env node
+
+import { readFile, access, constants } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import * as minimist from 'minimist';
+
+import { DocsWorkspace } from '../lib/markdown';
+import { URI } from 'vscode-uri';
+import { parse as parseYaml } from 'yaml';
+
+import type { HTML, Heading } from 'mdast';
+import type { Node, Literal } from 'unist';
+import type { visit as VisitFunction } from 'unist-util-visit';
+import type { fromHtml as FromHtmlFunction } from 'hast-util-from-html';
+import type { fromMarkdown as FromMarkdownFunction } from 'mdast-util-from-markdown';
+import { dynamicImport } from '../lib/helpers';
+import Ajv, { JSONSchemaType, ValidateFunction } from 'ajv';
+import AdmZip = require('adm-zip');
+
+// "<any char>: <match group>"
+const possibleStringRegex = /^[ \S]+?: *?(\S[ \S]+?)$/gm;
+const nonAlphaNumericDotRegex = /[^a-zA-Z0-9.]/g;
+
+interface ChangeSchema {
+  'pr-url': string;
+  'breaking-changes-header'?: string;
+  description?: string;
+}
+
+interface ApiHistory {
+  added?: ChangeSchema[];
+  deprecated?: ChangeSchema[];
+  changes?: ChangeSchema[];
+}
+
+interface Options {
+  checkPlacement: boolean;
+  checkPullRequestLinks: boolean;
+  breakingChangesFile: string;
+  checkStrings: boolean;
+  ignoreGlobs: string[];
+  schema: string;
+}
+
+interface HTMLWithPreviousNode extends HTML {
+  previousNode?: Node;
+}
+
+// If you change this, you might want to update the one in the website transformer
```

Comment:
Can we make these comments a bit more descriptive, so a person coming into this for the first time can better understand what the "website transformer" refers to? A blob link or explicitly naming the website repo is likely fine 🙂

Reply:
Since https://github.com/electron/website/pull/594 hasn't landed, I think it'd be okay to link the PR in this case!